front/src/Phaser/Game/GameMap.ts

```diff
@@ -8,15 +8,40 @@ export type PropertyChangeCallback = (newValue: string | number | boolean | unde
  * It is used to handle layer properties.
  */
 export class GameMap {
-    private key: number|undefined;
-    private lastProperties = new Map<string, string|boolean|number>();
+    private key: number | undefined;
+    private lastProperties = new Map<string, string | boolean | number>();
     private callbacks = new Map<string, Array<PropertyChangeCallback>>();
+
+    /**
+     * tileset.firstgid => (Map<tileid,Array<ITiledMapLayerProperty>>)
```

Comment:
maybe i should use an object here as you can specifiy it like so 

where the typedef is more helpful as it contains the key name

tileSetPropertyMap:{[tilset_firstgid:number]:{[tiled:number]:Array<ITiledMapLayerProperty>}}